internal/pkg/aws/elbv2/elbv2.go

```diff
@@ -73,6 +75,33 @@ func (e *ELBV2) ListenerRuleHostHeaders(ruleARN string) ([]string, error) {
 	return hostHeaders, nil
 }
 
+type Rule elbv2.Rule
+
+// DescribeRule returns the Rule with ruleARN.
+func (e *ELBV2) DescribeRule(ctx context.Context, ruleARN string) (Rule, error) {
+	resp, err := e.client.DescribeRules(&elbv2.DescribeRulesInput{
+		RuleArns: aws.StringSlice([]string{ruleARN}),
+	})
+	switch {
+	case err != nil:
+		return Rule{}, err
+	case len(resp.Rules) == 0:
+		return Rule{}, errors.New("not found")
+	}
```

Comment:
We probably don't need a switch here
```suggestion
        if err != nil:
            return Rule{}, fmt.Errorf("describe listener rules: %w", err)
        if !len(resp.Rules):
            return Rule{}, fmt.Errorf("cannot find any rules related to %s", ruleARN)
```

Reply:
sad

Reply:
lolll